### PR TITLE
feat: set up branch protection on GitHub

### LINKED ISSUE #7: set up branch protection

### PR DESCRIPTION
**Pull request by smira**
If Kres detects a git repo, finds link to GitHub upstream and if given
`GITHUB_API_TOKEN`, Kres sets up branch protection rule for the main
(`master`) branch of the project.

Fixes #7

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: internal/project/common/repository.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+func (r *Repository) CompileGitHub(client *github.Client) error {
+	branchProtection, resp, err := client.Repositories.GetBranchProtection(context.Background(), r.meta.GitHubOrganization, r.meta.GitHubRepository, r.MainBranch)
+	if err != nil {
+		if resp.StatusCode != http.StatusNotFound {
+			return err
+		}
+	}
+
+	req := github.ProtectionRequest{
+		RequireLinearHistory: github.Bool(true),
```
Comment (smira) [inline internal/project/common/repository.go:54]: this is the most important part: do we all agree that it's the settings we want?

File: internal/output/github/github.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	client *github.Client
+}
+
+// NewOutput creates new GitHub API output.
+func NewOutput() *Output {
+	output := &Output{}
+
+	token, exists := os.LookupEnv("GITHUB_API_TOKEN")
+	if !exists {
+		fmt.Println("GITHUB_API_TOKEN is missing, GitHub API integration is disabled")
```
Comment (andrewrynhard) [inline internal/output/github/github.go:28]: In the bot and in conform we use `GITHUB_TOKEN`. This env var is also respected by `hub` and the `gh` CLIs. Super nit, but I think maybe we should use that instead? Then my token that I have exported in my shell work, and it aligns with the official GH way.

File: internal/project/common/repository.go (hunk 1 above)
Comment (andrewrynhard) [reply internal/project/common/repository.go:54]: Plus one from me. Linear histories are awesome :smile:

File: internal/output/github/github.go (hunk 2 above)
Comment (smira) [reply internal/output/github/github.go]: updated, thanks!

**Comment by smira** [comment]
> I was thinking that _maybe_ this would be better in the bot code. Still not sure though. Thoughts?

I missed that comment, yep, we might move it there as well. The only benefit of Kres is that it's configurable easily per-repo vs. bot which doesn't have any config.

**Pull request merged**